src/model_repository_manager.cc

```diff
@@ -665,35 +697,69 @@ ModelRepositoryManager::LoadUnloadModels(
       }
 
       // Only update the infos when all validation is completed
+      ModelInfoMap current_infos;
+      CopyModelInfos(&current_infos);
       for (const auto& model_name : added) {
         auto nitr = new_infos.find(model_name);
-        infos_.emplace(model_name, std::move(nitr->second));
+        current_infos.emplace(model_name, std::move(nitr->second));
       }
       for (const auto& model_name : modified) {
         auto nitr = new_infos.find(model_name);
-        auto itr = infos_.find(model_name);
+        auto itr = current_infos.find(model_name);
         itr->second = std::move(nitr->second);
       }
+      new_infos.swap(current_infos);
```

Comment:
Yes, line 652 is not necessary. Updated.

Reply:
Line 652 is still required, because they are not at the same block. They look like this:
```
if (type == ActionType::UNLOAD) {
    ...skip...
    CopyModelInfos(&new_infos);  // line 652
    ...skip...
}
else {
    ...skip...
    new_infos.swap(current_infos);  // line 711
}
```
(removing line 652 will fail L0_lifecycle)

Reply:
Thanks for clarifying that, I missed the scope that the copy is in